lib/fluent/plugin/filter_viaq_data_model.rb

```diff
@@ -449,6 +462,40 @@ def transform_eventrouter(tag, record)
       end
     end
 
+    def handle_undefined_fields(tag, time, record)
+      if @undefined_to_string || @use_undefined || @undefined_dot_replace_char || (@undefined_max_num_fields > -1)
+        # undefined contains all of the fields not in keep_fields
+        undefined_keys = record.keys - @keep_fields.keys
+        return if undefined_keys.empty?
+        if @undefined_max_num_fields > -1 && undefined_keys.length > @undefined_max_num_fields
+          undefined = {}
+          undefined_keys.each{|k|undefined[k] = record.delete(k)}
+          record[@undefined_name] = JSON.dump(undefined)
```

Comment:
No need to check @use_undefined for this case?

Reply:
no - using `undefined_max_num_fields` implies that you want to use `undefined_name` - otherwise, there is no other place to put that value

Reply:
I updated the README